doc/help.md

```diff
@@ -580,49 +580,330 @@ PS1=$x.
 
 These builtins take input and output.  They're often used with redirects.
 
+<h4 id="read">read</h4>
+
+The read builtin reads lines from stdin and stores them into a variable. Input
+is split according to the $IFS delimiter.
+
+read FLAGS VAR+
```

Comment:
I think this should be `FLAGS* VARS*`

Because both flags and variables are optional.  I started using the regex of syntax of:

```
a*  0 or more
a+  1 or more
a?  0 or 1 (optional)
```

Reply:
Oooh, I noticed you were using them and tried to imitate it, but I didn't realize they were regex. That's a nice idea, will change it.